cumulus/pallets/xcmp-queue/src/lib.rs

```diff
@@ -1136,16 +1184,24 @@ impl<T: Config> InspectMessageQueues for Pallet<T> {
 
 		OutboundXcmpMessages::<T>::iter()
 			.map(|(para_id, _, messages)| {
-				let mut data = &messages[..];
-				let decoded_format = XcmpMessageFormat::decode(&mut data).unwrap();
-				if decoded_format != XcmpMessageFormat::ConcatenatedVersionedXcm {
-					panic!("Unexpected format.")
-				}
+				let data = &mut &messages[..];
+
+				let decoded_format = XcmpMessageFormat::decode(data).unwrap();
 				let mut decoded_messages = Vec::new();
 				while !data.is_empty() {
+					let message_bytes = match decoded_format {
+						XcmpMessageFormat::ConcatenatedVersionedXcm =>
+							Self::take_first_concatenated_xcm(data, &mut WeightMeter::new()),
+						XcmpMessageFormat::ConcatenatedOpaqueVersionedXcm =>
+							Self::take_first_concatenated_opaque_xcm(data),
+						_ => {
+							panic!("Unexpected format.")
```

Comment:
The messages in the `OutboundXcmpMessages` are guaranteed to be valid i guess?

Reply:
Yes, they should be valid, since the code in this pallet enqueues them.

But also this method is only used as part of a runtime API for dry-running XCMs and it's of to panic.